riscv/src/compiler.rs

```diff
@@ -1376,3 +1377,38 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use env_logger::filter;
+
+    use super::*;
+
+    #[test]
+    fn test_filter_matching_and_next_integers() {
+        assert_eq!(
+            filter_matching_and_next([0, 1, 2, 0, 2, 2, 1].iter(), |&&i| { i == 0 })
+                .map(|i| *i).collect::<Vec<_>>(),
+            vec![2, 2, 1]
+        );
+    }
```

Comment:
What should happen if we have two zeroes in a row?

Reply:
they are both removed, along with the following number, which is what we want: if the label is duplicated, we still want to remove the next statement.

Reply:
added that use case to the test btw